fix: improve performance with the ackedEvent store

jeremyletang opened the pull request:
Two things have been done here:
- first the buckets are for one hour instead of one per timestamps.
- hashes are stored in a map not a slice anymore to avoid long iterations

close #11497

jeremyletang committed:
fix: improve performance with the ackedEvent store

Signed-off-by: Jeremy Letang <me@jeremyletang.com>

wwestgarth commented:
In one of the previous panics about eth-call events being annoying, I lazily made a copy of `acked_events.go` in `core/datasource/external/ethverifier`...

Its probably worth copy+pasting this fix into that file too (or combining somehow if you can be bothered).

jeremyletang commented:
> In one of the previous panics about eth-call events being annoying, I lazily made a copy of `acked_events.go` in `core/datasource/external/ethverifier`...
> 
> Its probably worth copy+pasting this fix into that file too (or combining somehow if you can be bothered).

Addressed and rebased.
